tests/30rooms/13anonymousaccess.pl

```diff
@@ -385,6 +385,109 @@ sub check_events
       })
    };
 
+test "GET /publicRooms lists rooms",
+   requires => [qw( first_api_client ), local_user_fixture() ],
+
+   check => sub {
+      my ( $http, $user ) = @_;
+
+      Future->needs_all(
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest0",
+         ),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest1",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_history_visibility( $user, $room_id, "world_readable" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest2",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_history_visibility( $user, $room_id, "invited" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest3",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_guest_access( $user, $room_id, "can_join" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest4",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            Future->needs_all(
+               matrix_set_room_guest_access( $user, $room_id, "can_join" ),
+               matrix_set_room_history_visibility( $user, $room_id, "world_readable" ),
+            );
+         }),
+      )->then( sub {
+         $http->do_request_json(
+            method => "GET",
+            uri    => "/api/v1/publicRooms",
+         )->then( sub {
+            my ( $body ) = @_;
+
+            log_if_fail "publicRooms", $body;
+
+            require_json_keys( $body, qw( start end chunk ));
+            require_json_list( $body->{chunk} );
+
+            my %seen = (
+               listingtest0 => 0,
+               listingtest1 => 0,
+               listingtest2 => 0,
+               listingtest3 => 0,
+               listingtest4 => 0,
+            );
+
+            foreach my $room ( @{ $body->{chunk} } ) {
+               my $aliases = $room->{aliases};
+               my $world_readable = $room->{world_readable};
+               my $guest_can_join = $room->{guest_can_join};
+
+               foreach my $alias ( @{$aliases} ) {
+                  if( $alias =~ m/^\Q#listingtest0:/ ) {
+                     $seen{listingtest0} = $world_readable == JSON::false && $guest_can_join == JSON::false;
```

Comment:
So contrary to my earlier statements, `==` comparison on JSON booleans isn't really DWIM'ing here. This is just doing boolean truth comparison, because the JSON values helpfully overload the `==` operator. Since we're really requiring on the API that these really honestly be true JSON booleans, you might want to put

    require_json_boolean( my $world_readable = $room->{world_readable} );

at the top of the loop, and just test on simple boolean truth later on; e.g.

    $seen{listeningtest0} = !$world_readable && !$guest_can_join;